pkg/deployments/tasks/20230314-batch-relayer-v5/test/composableStableV2.fork.ts

```diff
@@ -0,0 +1,279 @@
+
+        const { tokens: registeredTokens } = await vault.getPoolTokens(poolId);
+        // Given the bptOut, the max amounts in should be slightly more than 1/5. Decimals make it a bit complicated.
+        const adjustedBalances = [
+          initialBalanceDAI.div(fp(4.99)).mul(fp(1)),
+          initialBalanceUSDC.div(bn(4.99e6)).mul(1e6),
+        ];
+        const maxAmountsIn = getRegisteredBalances(bptIndex, adjustedBalances);
+
+        const tx = await vault.connect(whale).joinPool(poolId, whale.address, whale.address, {
+          assets: registeredTokens,
+          maxAmountsIn: maxAmountsIn,
+          fromInternalBalance: false,
+          userData: StablePoolEncoder.joinAllTokensInForExactBptOut(bptOut),
+        });
+        const receipt = await (await tx).wait();
+        const { deltas: amountsIn } = expectEvent.inReceipt(receipt, 'PoolBalanceChanged').args;
+
+        // Amounts in should be ~ 1/5 the initial balances
+        expect(amountsIn).to.equalWithError(maxAmountsIn, 0.01);
+
+        // Make sure received BPT is close to what we expect
+        const currentBptBalance = await pool.balanceOf(whale.address);
+        expect(currentBptBalance).to.be.equalWithError(bptOut, 0.001);
+      });
+    });
+
+    describe('proportional exit (outside relayer)', () => {
+      before('deploy pool', async () => {
+        pool = await createPool();
+
+        poolId = await pool.getPoolId();
+        const [registeredAddress] = await vault.getPool(poolId);
+        expect(registeredAddress).to.equal(pool.address);
+
+        bptIndex = await pool.getBptIndex();
+      });
+
+      it('exits proportionally', async () => {
+        const previousBptBalance = await pool.balanceOf(owner.address);
+        const bptIn = previousBptBalance.div(4);
+
+        const { tokens: registeredTokens, balances: registeredBalances } = await vault.getPoolTokens(poolId);
+
+        const tx = await vault.connect(owner).exitPool(poolId, owner.address, owner.address, {
+          assets: registeredTokens,
+          minAmountsOut: Array(registeredTokens.length).fill(0),
+          fromInternalBalance: false,
+          userData: StablePoolEncoder.exitExactBptInForTokensOut(bptIn),
+        });
+        const receipt = await (await tx).wait();
+        const { deltas } = expectEvent.inReceipt(receipt, 'PoolBalanceChanged').args;
+        const amountsOut = deltas.map((x: BigNumber) => x.mul(-1));
+
+        const expectedAmountsOut = (registeredBalances as BigNumber[]).map((b) => b.div(4));
+        expectedAmountsOut[bptIndex] = bn(0);
+
+        // Amounts out should be 1/4 the initial balances
+        expect(amountsOut).to.equalWithError(expectedAmountsOut, 0.00001);
+
+        // Make sure sent BPT is close to what we expect
+        const currentBptBalance = await pool.balanceOf(owner.address);
+        expect(currentBptBalance).to.be.equalWithError(bn(previousBptBalance).sub(bptIn), 0.001);
+      });
+    });
```

Comment:
Why have these? They are unrelated to the relayer

Reply:
Can remove; like the other comments. Just had them because that's what I did first, to test that everything up to the relayer was working, before testing the relayer component, so that I didn't have to look in two places for any issues.

Reply:
That makes sense, but I think keeping them in the test doesn't add much and makes the thing overall a bit confusing

Reply:
Ok. I did remove them.